File: server-scala-play-server/app/model/ApplicationResponse.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for ApplicationResponse.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class ApplicationResponse(
  valid: Option[Boolean],
  message: Option[String],
  version: Option[Double],
  serializeNulls: Option[Boolean],
  startTimeLog: Option[Long],
  errorCode: Option[String],
  request: Option[List[NameStringValueResponse]],
  applicationId: Option[Long],
  appKey: Option[String],
  title: Option[String],
  currentBuildVersion: Option[String],
  currentApiVersion: Option[String],
  appIcon: Option[AssetShortResponse],
  appLogo: Option[AssetShortResponse],
  active: Option[Boolean],
  about: Option[String],
  landingPageUrl: Option[String],
  landingPageFullUrl: Option[String],
  defaultImageFullUrl: Option[String],
  activityDescription: Option[String],
  platforms: Option[List[PlatformResponse]],
  categories: Option[List[CategoryResponse]],
  placementCount: Option[Int],
  billable: Option[BillableEntityShortResponse],
  hasAds: Option[Boolean],
  publicNotifications: Option[Boolean],
  urlScheme: Option[String],
  global: Option[Boolean],
  downloadUrls: Option[String],
  localAdsEnabled: Option[Boolean],
  trilatSettings: Option[TrilatAppSettings],
  appType: Option[String],
  appName: Option[String],
  bundleId: Option[String],
  scoringType: Option[ApplicationResponse.ScoringType.Value],
  hintCost: Option[Int],
  maxScore: Option[Int],
  ticketsPerPoint: Option[Float],
  hasGameData: Option[Boolean],
  globalTickets: Option[Boolean],
  eulaVersion: Option[String],
  eulaDateUpdated: Option[Long],
  modules: Option[String],
  billableEntityName: Option[String],
  showInActivities: Option[Boolean],
  cssClass: Option[String],
  inviteWelcomeText: Option[String],
  invitePageUrl: Option[String],
  apnsCertificateFileName: Option[String],
  facebookAppId: Option[String],
  facebookAppSecret: Option[String],
  appToken: Option[String],
  googleApiKey: Option[String],
  appRestKey: Option[String],
  publicContentApproval: Option[Boolean],
  productionMode: Option[Boolean],
  minimumSessionLength: Option[Int],
  sessionGapLength: Option[Int],
  sqootApiKey: Option[String],
  amqpEnabled: Option[Boolean],
  authorizedCount: Option[Int],
  authorizedServers: Option[String],
  returning: Option[String]
)

object ApplicationResponse {
  implicit lazy val applicationResponseJsonFormat: Format[ApplicationResponse] = Json.format[ApplicationResponse]

  // noinspection TypeAnnotation
  object ScoringType extends Enumeration {
    val GAMELEVEL = Value("GAME_LEVEL")
    val GAMEOBJECT = Value("GAME_OBJECT")

    type ScoringType = Value
    implicit lazy val ScoringTypeJsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
  }
}

File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/ApplicationResponse.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class ApplicationResponse 
{
    public bool Valid { get; set; }
    public string Message { get; set; }
    public double VarVersion { get; set; }
    public bool SerializeNulls { get; set; }
    public long StartTimeLog { get; set; }
    public string ErrorCode { get; set; }
    public List<NameStringValueResponse> Request { get; set; }
    public long ApplicationId { get; set; }
    public string AppKey { get; set; }
    public string Title { get; set; }
    public string CurrentBuildVersion { get; set; }
    public string CurrentApiVersion { get; set; }
    public AssetShortResponse AppIcon { get; set; }
    public AssetShortResponse AppLogo { get; set; }
    public bool Active { get; set; }
    public string About { get; set; }
    public string LandingPageUrl { get; set; }
    public string LandingPageFullUrl { get; set; }
    public string DefaultImageFullUrl { get; set; }
    public string ActivityDescription { get; set; }
    public List<PlatformResponse> Platforms { get; set; }
    public List<CategoryResponse> Categories { get; set; }
    public int PlacementCount { get; set; }
    public BillableEntityShortResponse Billable { get; set; }
    public bool HasAds { get; set; }
    public bool PublicNotifications { get; set; }
    public string UrlScheme { get; set; }
    public bool Global { get; set; }
    public string DownloadUrls { get; set; }
    public bool LocalAdsEnabled { get; set; }
    public TrilatAppSettings TrilatSettings { get; set; }
    public string AppType { get; set; }
    public string AppName { get; set; }
    public string BundleId { get; set; }
    
    /// <summary>
    /// Gets or Sets ScoringType
    /// </summary>
    public enum ScoringTypeEnum
    {
        
        /// <summary>
        /// Enum GAMELEVELEnum for GAME_LEVEL
        /// </summary>
        GAMELEVELEnum = 1,
        
        /// <summary>
        /// Enum GAMEOBJECTEnum for GAME_OBJECT
        /// </summary>
        GAMEOBJECTEnum = 2
    }

    public ScoringTypeEnum ScoringType { get; set; }
    public int HintCost { get; set; }
    public int MaxScore { get; set; }
    public float TicketsPerPoint { get; set; }
    public bool HasGameData { get; set; }
    public bool GlobalTickets { get; set; }
    public string EulaVersion { get; set; }
    public long EulaDateUpdated { get; set; }
    public string Modules { get; set; }
    public string BillableEntityName { get; set; }
    public bool ShowInActivities { get; set; }
    public string CssClass { get; set; }
    public string InviteWelcomeText { get; set; }
    public string InvitePageUrl { get; set; }
    public string ApnsCertificateFileName { get; set; }
    public string FacebookAppId { get; set; }
    public string FacebookAppSecret { get; set; }
    public string AppToken { get; set; }
    public string GoogleApiKey { get; set; }
    public string AppRestKey { get; set; }
    public bool PublicContentApproval { get; set; }
    public bool ProductionMode { get; set; }
    public int MinimumSessionLength { get; set; }
    public int SessionGapLength { get; set; }
    public string SqootApiKey { get; set; }
    public bool AmqpEnabled { get; set; }
    public int AuthorizedCount { get; set; }
    public string AuthorizedServers { get; set; }
    public string Returning { get; set; }
}


